package/gluon-config-mode-outdoor/luasrc/lib/gluon/config-mode/wizard/0250-outdoor.lua

```diff
@@ -0,0 +1,28 @@
+
+return function(form, uci)
+	local outdoor_mode = uci:get_bool('gluon', 'wireless', 'outdoor')
+
+	if not outdoor_mode then
+		-- only shown when enabled. enabled through preset or web-wifi-config.
+		return
+	end
+
+	local pkg_i18n = i18n 'gluon-config-mode-outdoor'
+
+	local section = form:section(Section, nil, pkg_i18n.translate(
+		"Please enable this option in case the node is to be installed outdoors "
+		.. "to comply with local frequency regulations."
+	))
+
+	local outdoor = section:option(Flag, 'outdoor', pkg_i18n.translate("Node will be installed outdoors"))
+	outdoor.default = outdoor_mode
+
+	function outdoor:write(data)
+		if data ~= outdoor_mode then
+			uci:set('gluon', 'wireless', 'outdoor', data)
+			uci:commit('gluon.wireless')
+			os.execute('/lib/gluon/upgrade/200-wireless')
+			uci:commit('wireless')
+		end
+	end
```

Comment:
- Instead of commiting the data in outdoor:write, return `{'gluon', 'wireless'}` from the outer function.
- Section-specific commits like `gluon.wireless` are not useful. The file will be written anyways.

Reply:
I need to commit gluon before running `200-wireless` or else it will not apply outdoor settings.

Reply:
`uci:save('gluon')` should be sufficient before running the upgrade script, the actual commit can then be done via the return.

Reply:
Addressed in df0ce8f4ea9357dd223d97a393911bedcaee8852